backend/src/main/java/ai/verta/modeldb/reconcilers/UpdateParentTimestampReconcile.java

```diff
@@ -0,0 +1,73 @@
+package ai.verta.modeldb.reconcilers;
+
+import ai.verta.modeldb.common.futures.FutureJdbi;
+import ai.verta.modeldb.common.reconcilers.ReconcileResult;
+import ai.verta.modeldb.common.reconcilers.Reconciler;
+import ai.verta.modeldb.common.reconcilers.ReconcilerConfig;
+import java.util.AbstractMap;
+import java.util.Set;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class UpdateParentTimestampReconcile
+    extends Reconciler<AbstractMap.SimpleEntry<Long, Long>> {
+  private static final Logger LOGGER = LogManager.getLogger(UpdateParentTimestampReconcile.class);
+  private final FutureJdbi futureJdbi;
+
+  public UpdateParentTimestampReconcile(ReconcilerConfig config, FutureJdbi futureJdbi) {
+    super(config, LOGGER);
+    this.futureJdbi = futureJdbi;
+  }
+
+  @Override
+  public void resync() {
+    var fetchUpdatedDatasetIds =
+        new StringBuilder("SELECT rc.repository_id, MAX(cm.date_created) AS max_date ")
+            .append(" FROM commit cm INNER JOIN repository_commit rc ")
+            .append(" ON rc.commit_hash = cm.commit_hash ")
+            .append(" INNER JOIN commit_parent cp ")
+            .append(" ON cp.parent_hash IS NOT NULL ")
+            .append(" AND cp.child_hash = cm.commit_hash ")
+            .append(" INNER JOIN repository rp ")
+            .append(" ON rp.id = rc.repository_id AND rp.date_updated < cm.date_created ")
+            .append(" GROUP BY rc.repository_id")
+            .toString();
+
+    futureJdbi.useHandle(
+        handle -> {
+          handle
+              .createQuery(fetchUpdatedDatasetIds)
+              .setFetchSize(config.maxSync)
+              .map(
+                  (rs, ctx) -> {
+                    Long datasetId = rs.getLong("rc.repository_id");
+                    Long maxUpdatedDate = rs.getLong("max_date");
+                    this.insert(new AbstractMap.SimpleEntry<>(datasetId, maxUpdatedDate));
+                    return rs;
+                  })
+              .list();
+        });
+  }
+
+  @Override
+  protected ReconcileResult reconcile(Set<AbstractMap.SimpleEntry<Long, Long>> updatedMaxDateMap) {
+    LOGGER.debug("Reconciling parent time update for repository" + updatedMaxDateMap.size());
+
+    futureJdbi.useHandle(
```

Comment:
This will create a loose future, which can add a lot of pressure to the DB. We had the same issue in another reconciler. Could you copy the solution from there?

Reply:
Fixed